lib/lambda_ethereum_consensus/p2p/peerbook.ex

```diff
@@ -35,10 +45,54 @@ defmodule LambdaEthereumConsensus.P2P.Peerbook do
     {:reply, peer_id, peerbook}
   end
 
+  @impl true
+  def handle_cast({:remove_peer, peer_id}, peerbook) do
+    updated_peerbook = Map.delete(peerbook, peer_id)
+    {:noreply, updated_peerbook}
+  end
+
   @impl true
   def handle_info({:new_peer, peer_id}, peerbook) do
     :telemetry.execute([:peers, :connection], %{id: peer_id}, %{result: "success"})
     updated_peerbook = Map.put(peerbook, peer_id, @initial_score)
     {:noreply, updated_peerbook}
   end
+
+  @impl true
+  def handle_info(:prune, peerbook) do
+    len = map_size(peerbook)
+
+    prune_size =
+      (len * @prune_percentage)
+      |> round()
+      |> min(@max_prune_size)
+      |> min(len - @target_peers)
+      |> max(0)
+
+    n = :rand.uniform(len)
+
+    peerbook
+    |> Map.keys()
+    |> Stream.drop(n)
+    |> Stream.take(prune_size)
+    |> Enum.each(fn peer_id -> Task.start(__MODULE__, :challenge_peer, [peer_id]) end)
+
+    schedule_pruning()
+    {:noreply, peerbook}
+  end
+
+  def challenge_peer(peer_id) do
```

Comment:
Is this logic taken from another client or was this your idea?

Reply:
It was my idea, although other clients probably do the same. Requesting metadata and status data from peers is the best way to filter them based on syncing status.